crates/symbolicator/src/endpoints/minidump.rs

```diff
@@ -32,17 +32,34 @@ async fn handle_minidump_request(
 
         let content_disposition = field.content_disposition();
         match content_disposition.as_ref().and_then(|d| d.get_name()) {
-            Some("upload_file_minidump") => minidump = Some(read_multipart_file(field).await?),
+            Some("upload_file_minidump") => {
+                let minidump_file = tempfile::Builder::new()
+                    .prefix("minidump")
+                    .suffix(".dmp")
+                    .tempfile()?;
+                let (mut file, temp_path) = minidump_file.into_parts();
+
+                let mut stream = field.compat();
+                // it would have been nice to use `tokio::io::copy`, or at least the non-blocking
+                // `tokio::fs::File` / `tokio::io::AsyncWriteExt`, but that needs to run in the
+                // context of a tokio 1 executor, which these futures do not.
+                //let mut file = tokio::fs::File::from_std(file);
+                while let Some(chunk) = stream.next().await {
+                    file.write_all(&chunk?)?;
+                }
```

Comment:
This is basically the intent of `read_multipart_file`. It would be nice to move this logic there so it's used by all places dealing with files.

The contents of the comment here are true, although I'm not sure if it's worth writing such a long rant for such a simple loop.